src/renderer/components/configure-recording-set-page/index.tsx

```diff
@@ -115,154 +47,185 @@ const ConfigureRecordingSetPage: FC<{
+        const states: MetadataState[] = [];
+        if (isSubscribed) {
+          if (responses[0] == 'file') {
+            states.push('list-preview');
+            setFilePathRec(listFilePath);
+          } else {
+            setFilePathRec('');
+          }
+          if (responses[1] == 'file') {
+            states.push('oto-ini');
+            setFilePathOto(otoFilePath);
+          } else {
+            setFilePathOto('');
+          }
+          if (responses[2] == 'file') {
+            states.push('dvcfg');
+            setFilePathDvcfg(dvcfgFilePath);
+          } else {
+            setFilePathDvcfg('');
+          }
+          if (states.length == 0) {
+            setMetadataStateIndex(-1);
+          } else {
+            setDropdownStateArray(states);
+            setMetadataStateIndex(0);
+          }
+        }
+      })
+      .catch((error) => {
+        log.error(error);
+        throw error;
+      });
+    return () => {
+      isSubscribed = false;
+    };
   };
 
-  useEffect(() => {
-    onSetSelected(recordingSets[selectedRecordingSetIndex]);
-    // We have to ignore onSetSelected because it's new every time
-    // eslint-disable-next-line
-  }, [recordingSets, selectedRecordingSetIndex]);
-
-  useEffect(() => {
-    if (canWrite && recordingProject) {
-      const currentProjectState: Partial<ProjectFile> = {
-        name: recordingProject.name,
-        recordingSets,
-      };
-      writeFile(
-        join(recordingProject.rootPath, PROJECT_CONFIG_FILENAME),
-        JSON.stringify(currentProjectState, null, 2),
-      ).catch(log.error);
+  switch (recordingSetState) {
+    case 'home':
+      return (
+        <ConfigureRecordingSet
+          onNext={onNext}
+          onBack={onBack}
+          onSetSelected={onSetSelected}
+          setRecordingSetState={setRecordingSetState}
+          prevState={prevState}
+          currState={recordingSetState}
+          chosenBuiltInList={chosenBuiltInList}
+          rawSetChosenBuiltInList={rawSetChosenBuiltInList}
+          chosenCustomListPath={chosenCustomListPath}
+          rawSetChosenCustomListPath={rawSetChosenCustomListPath}
+          metaDataIndex={metadataStateIndex}
+          getFilePath={updatePaths}
+        />
+      );
+    case 'external':
+      return (
+        <ConfigureRecordingSet
+          onNext={onNext}
+          onBack={onBack}
+          onSetSelected={onSetSelected}
+          setRecordingSetState={setRecordingSetState}
+          prevState={prevState}
+          currState={recordingSetState}
+          chosenBuiltInList={chosenBuiltInList}
+          rawSetChosenBuiltInList={rawSetChosenBuiltInList}
+          chosenCustomListPath={chosenCustomListPath}
+          rawSetChosenCustomListPath={rawSetChosenCustomListPath}
+          metaDataIndex={metadataStateIndex}
+          getFilePath={updatePaths}
+        />
+      );
+    case 'metadata':
+      if (dropDownStates[metadataStateIndex] == 'list-preview') {
```

Comment:
Why not extract `dropDownStates[metadataStateIndex]` into a variable?

Reply:
Also, this can be a switch statement too

Reply:
I'm using an array to make it easy to circle though the different local states - the preview page can just call the function that circularly increments the array instead of having to do more complex logic to determine what the next page is

Reply:
I meant you could just do `const currentState = dropDownStates[metadataStateIndex]` above the statements.